src/libs/actions/WelcomeActions.js

```diff
@@ -9,75 +8,42 @@ import ONYXKEYS from '../../ONYXKEYS';
-Onyx.connect({
-    key: ONYXKEYS.NVP_IS_FIRST_TIME_NEW_EXPENSIFY_USER,
-    callback: val => isFirstTimeNewExpensifyUser = val,
-});
-
-const allReports = {};
-Onyx.connect({
-    key: ONYXKEYS.COLLECTION.REPORT,
-    callback: (val, key) => {
-        if (!val || !key) {
-            return;
-        }
-
-        allReports[key] = {...allReports[key], ...val};
-    },
-});
-
-const allPolicies = {};
-Onyx.connect({
-    key: ONYXKEYS.COLLECTION.REPORT,
-    callback: (val, key) => {
-        if (!val || !key) {
-            return;
-        }
-
-        allPolicies[key] = {...allPolicies[key], ...val};
-    },
-});
-
 /**
  * Shows a welcome action on first login
  *
  * @param {Object} params
  * @param {Object} params.routes
- * @param {Function} params.hideCreateMenu
+ * @param {Function} params.showCreateMenu
  */
-function show({routes, hideCreateMenu}) {
-    // NOTE: This setTimeout is required due to a bug in react-navigation where modals do not display properly in a drawerContent
-    // This is a short-term workaround, see this issue for updates on a long-term solution: https://github.com/Expensify/App/issues/5296
-    setTimeout(() => {
-        if (!isFirstTimeNewExpensifyUser) {
-            return;
-        }
-
-        // Set the NVP back to false so we don't automatically run welcome actions again
-        NameValuePair.set(CONST.NVP.IS_FIRST_TIME_NEW_EXPENSIFY_USER, false, ONYXKEYS.NVP_IS_FIRST_TIME_NEW_EXPENSIFY_USER);
-
-        // We want to display the Workspace chat first since that means a user is already in a Workspace and doesn't need to create another one
-        const workspaceChatReport = _.find(allReports, report => ReportUtils.isPolicyExpenseChat(report));
-        if (workspaceChatReport) {
-            Navigation.navigate(ROUTES.getReportRoute(workspaceChatReport.reportID));
-            return;
-        }
-
-        // If we are rendering the SidebarScreen at the same time as a workspace route that means we've already created a workspace via workspace/new and should not open the global
-        // create menu right now.
-        const topRouteName = lodashGet(_.last(routes), 'name', '');
-        const loginWithShortLivedTokenRoute = _.find(routes, route => route.name === 'LogInWithShortLivedToken');
-        const exitingToWorkspaceRoute = lodashGet(loginWithShortLivedTokenRoute, 'params.exitTo', '') === 'workspace/new';
-        const isDisplayingWorkspaceRoute = topRouteName.toLowerCase().includes('workspace') || exitingToWorkspaceRoute;
-
-        // It's also possible that we already have a workspace policy. In either case we will not hide the menu but do still want to set the NVP in this case
-        // since the user does not need to create a workspace.
-        if (!Policy.isAdminOfFreePolicy(allPolicies) && !isDisplayingWorkspaceRoute) {
-            hideCreateMenu();
-        }
-    }, 1500);
+function show({
+    routes, showCreateMenu, isFirstTimeNewExpensifyUser, allReports, allPolicies,
+}) {
```

Comment:
nab; is a lint rule enforcing doing this on multiple lines? Or personal preference? My personal preference is to keep this 1 line 😅 but def not super worried about it

Reply:
yeah, it was the linter. I don't have any strong feelings on this though - we can change the linter if it bothers enough people 😄